src/Sarif.WorkItemFiling/Filtering/IFilteringStrategy.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItemFiling.Filtering
+{
+    /// <summary>
+    /// Interface that defines a strategy for selecting only those SARIF results that should
+    /// be filed as work items.
+    /// </summary>
+    public interface IFilteringStrategy
```

Comment:
>IFilteringStrategy [](start = 21, length = 18)

What is your rationale for preferring an interface in this case over an abstract in FilingTarget (soon to be WorkItemHost, I hope).

Reply:
My rationale was poor. I was thinking of abstract base class as a place to put shared implementation helpers, and I couldn't imagine there being any such for the strategy classes. But:

1. The stronger rationale for abstract base classes is that it allows you to add new _members_ to the base, without breaking clients.
2. Uniformity is a good thing, and if we use abstract base classes everywhere, then we don't have readers of the code wondering "why here but not there?".

Fixed

---
In reply to: [306115581](https://github.com/microsoft/sarif-sdk/pull/1582#discussion_r306115581) [](ancestors = 306115581)